### PR TITLE
19 create a script for to generate the checksum

### LINKED ISSUE #19: create a script for to generate the checksum
**Context**
as a dev when I want to list my plugin on the plugin store of massa station, I have to generate checksum of my plugin based on the content of the artifacts  and I will be able to do it automatically using this script. 

**User flow**
*Describe the user flow using user stories so the end result is super clear - PM*
the dev has to :
- fill the plugins.json with the right informations, he can skip the checksum fields
- run a script to generate & write the appropriate checksums

**How to**
write a script to generate the checksum of the plugin's artifacts 

**Technical details**
the functions to calculate the checksums are defined in the test
**QA testing**
Update a plugin with the script & create a pr.
run `npm write-checksum massa-wallet` or / and `npm write-checksum --all`
run `npm run test`

### PR DESCRIPTION
**Pull request by Pierregvx**
[Screencast from 11-07-2023 16:59:14.webm](https://github.com/massalabs/station-store/assets/83134981/cbee9332-a332-473d-a9a8-67beef150f71)

### CODE REVIEW

**Review by veratek — APPROVED** [review]

File: src/write-checksum.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { StorePlugin, calculateChecksum, getPluginsData } from './utils';
+import { format } from 'prettier';
+
+async function main(name?: string) {
+  const plugins: StorePlugin[] = await getPluginsData();
+  
+  if (name) {
+    const plugin = plugins.find(
+      (plugin) =>
+        slugify(plugin.name, { lower: true }) === slugify(name, { lower: true })
```
Comment (mazmassa) [inline src/write-checksum.ts]: Are we sure we'll add a whooooole lib to to that!? A simple regex would not be enough?

File: src/write-checksum.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+async function processPlugin(plugin: StorePlugin) {
+  const { assets } = plugin;
+  await Promise.all(
+    Object.keys(assets).map(async (key) => {
+      const asset = assets[key];
+      const checksum = await calculateChecksum(asset.url);
+      plugin.assets[key].checksum = checksum;
+    })
```
Comment (mazmassa) [inline src/write-checksum.ts]: I am having a hard time to understand here. You are waiting for all promises to be resolved in Promisse.all. Why not have the checksum returned from the block and avoid the nested async/await.

Comment (Pierregvx) [reply src/write-checksum.ts]: I removed it here

File: src/write-checksum.ts (hunk 1 above)
Comment (Pierregvx) [reply src/write-checksum.ts]: hmm I am using slugify because I am using this script with the cli
How would you use the regex ?

Comment (mazmassa) [reply src/write-checksum.ts]: Im not sure I am following you. What are you looking for when use slugify? Turn a name-of-something in name_of_something!?

File: src/write-checksum.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const formattedPluginsJson = format(pluginsJson, { parser: 'json' });
+  fs.writeFileSync('plugins.json', formattedPluginsJson);
+}
+
+async function processPlugin(plugin: StorePlugin) {
+  const { assets } = plugin;
+  for (const key of Object.keys(assets)) {
+    const asset = assets[key];
+    const checksum = await calculateChecksum(asset.url);
+    asset.checksum = checksum;
```
Comment (mazmassa) [inline src/write-checksum.ts]:
```suggestion
    asset.checksum = await calculateChecksum(asset.url);
```

File: src/write-checksum.ts (hunk 1 above)
Comment (Pierregvx) [reply src/write-checksum.ts]: If I want to be able to find `Massa Wallet` when I am running `npm run massa-wallet`

Comment (mazmassa) [reply src/write-checksum.ts]: No, I mean in this line:

```slugify(plugin.name, { lower: true }) === slugify(name, { lower: true })```

Are you trying to transform `Massa Wallet` into `massa-wallet` to be able to compare them!? Right!?

If yes, you definitely don't need a whooooole lib to do that!

Comment (mazmassa) [reply src/write-checksum.ts]:
```ts
function slugify (str: string): string {
  return str.trim().toLowerCase().replace(/\s+/g, '-');
}

console.log(slugify('Something Else'));
// something-else
```

Comment (Pierregvx) [reply src/write-checksum.ts]: I see, thanks!

File: src/write-checksum.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  const pluginsJson = JSON.stringify(plugins, null, 2);
+  const formattedPluginsJson = format(pluginsJson, { parser: 'json' });
+  fs.writeFileSync('plugins.json', formattedPluginsJson);
+}
+
+async function processPlugin(plugin: StorePlugin) {
+  const { assets } = plugin;
+  for (const key of Object.keys(assets)) {
```
Comment (mazmassa) [inline src/write-checksum.ts:35]:
```suggestion

  for (const key of Object.keys(assets)) {
```

**Review by mazmassa — APPROVED** [review]

**Pull request merged**